pkg/deployments/manager.go

```diff
@@ -65,7 +65,11 @@ func (r *Manager) SetDesiredScale(deploymentName string, n int32) {
 
 func (r *Manager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	var d appsv1.Deployment
-	if err := r.Get(ctx, req.NamespacedName, &d); err != nil {
+	switch err := r.Get(ctx, req.NamespacedName, &d); {
+	case apierrors.IsNotFound(err):
```

Comment:
Reconcile is also called when a deployment is deleted. The name is unknown to the k8s api so we know it is deleted